### PR TITLE
Fix/fullpath

### PR DESCRIPTION
**Pull request by KostiaLevchenko**
- fixed the appearance of extra slashes in fpath
- fixed stack smashing while fpath too long

### CODE REVIEW

File: src/kfs_lib/includes/kfs_common.h [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #ifndef KFS_COMMON
 #define KFS_COMMON
+#define COPYMODE        0644
+#define BUFFERSIZE      4096
+#define MAX_PATH        8192
```
Comment (kopachevsky) [inline src/kfs_lib/includes/kfs_common.h]: PATH_PATH_EXTENDED

Comment (KostiaLevchenko) [reply src/kfs_lib/includes/kfs_common.h]: fixed in 7faaab1

File: src/kfs_lib/includes/kfs_common.h [hunk 2]
```diff
@@ -1,5 +1,9 @@
 #ifndef KFS_COMMON
 #define KFS_COMMON
+#define COPYMODE        0644
+#define BUFFERSIZE      4096
+#define PATH_MAX_EXTENDED        8192   //extended version of PATH MAX from limits.h (doubled size )
```
Comment (kopachevsky) [inline src/kfs_lib/includes/kfs_common.h]: it's not good to use direct value? why not use MAX_PATH*2?

File: src/kfs_lib/includes/kfs_common.h [hunk 3]
```diff
@@ -1,5 +1,9 @@
 #ifndef KFS_COMMON
 #define KFS_COMMON
+#define COPYMODE        0644
+#define BUFFERSIZE      4096
```
Comment (kopachevsky) [inline src/kfs_lib/includes/kfs_common.h:4]: what this for?

Comment (KostiaLevchenko) [reply src/kfs_lib/includes/kfs_common.h:4]: for copy cluster content functions

File: src/kfs_lib/includes/kfs_common.h (hunk 2 above)
Comment (KostiaLevchenko) [reply src/kfs_lib/includes/kfs_common.h]: fixed 399cd8a

**Pull request merged**